pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	ScrapeConfigKind    = "ScrapeConfig"
+	ScrapeConfigName    = "scrapeconfigs"
+	ScrapeConfigKindKey = "scrapeconfig"
+)
+
+// +genclient
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:categories="prometheus-operator",shortName="scfg"
+// +kubebuilder:storageversion
+
+// ScrapeConfig defines a namespace ScrapeConfig to be aggregated across multiple namespaces inta  prometheus
+// configuration
+type ScrapeConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ScrapeConfigSpec `json:"spec"`
+}
+
+// DeepCopyObject implements the runtime.Object interface.
+func (l *ScrapeConfig) DeepCopyObject() runtime.Object {
+	return l.DeepCopy()
+}
+
+// ScrapeConfigList is a list of ScrapeConfigs.
+// +k8s:openapi-gen=true
+type ScrapeConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata
+	// More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// List of ScrapeConfigs
+	Items []*ScrapeConfig `json:"items"`
+}
+
+// DeepCopyObject implements the runtime.Object interface.
+// TODO: understand what this does
```

Comment:
As the comment above says, it's required to implement the runtime.Object interface.
`ScrapeConfig` and `ScrapeConfigList` should be added there too:
https://github.com/prometheus-operator/prometheus-operator/blob/8c73904adee24013d8800336554d8709a565fb4a/pkg/apis/monitoring/v1alpha1/register.go#L49

Reply:
Thanks!

Reply:
Fixed in 8cdf5623e